### PR TITLE
fix: applicability for unique id rule

### PR DESCRIPTION
**Pull request by jkodu**
Update applicability

Closes issue: 
https://github.com/auto-wcag/auto-wcag/issues/349

## Guidance for the PR (pull request) creator

**When creating PR:**
- [ ] Make sure you requesting to **pull a issue/feature/bugfix branch** (right side) to the master branch (left side).

**After creating PR:**
- [ ] Add yourself (and co-authors) as "Assignees" for PR
- [ ] Add label to indicate if it's a `Rule`, `Definition` or `Chore` (more to the administrative side)
- [ ] Add relevant project (e.g. "Q3 2018 Status") to PR
- [ ] OPTIONAL: If you want anyone in particular to review your pull request, assign them as "Reviewers".
- [ ] Close the issue that the PR resolves (and make sure the issue is referenced in the top of this comment)

# How to Review And Approve
- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines. 
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.

### CODE REVIEW

File: _rules/SC4-1-1-unique-id.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ authors:
 
 ### Applicability
 
-Any `id` attribute specified on an HTML or SVG element.
+Any [non-empty](#non-empty) `id` attribute specified on an HTML or SVG element.
```
Comment (WilcoFiers) [inline _rules/SC4-1-1-unique-id.md]: Non-empty isn't the right definition here. I can't really tell what you were trying to do here because you didn't add any test cases. But I doubt `id="-"` should be ignored by this, which is what "non-empty" does.

Comment (annethyme) [reply _rules/SC4-1-1-unique-id.md]: We discussed whether it would make sense to split the "non-empty" definition out into two different definitions: 

1. Non-empty: A string of characters is considered non-empty if the string contains characters that are not space characters. (or even have the definition be of "empty" instead)

2. Textual content: A string of characters is considered textual content if it contains one or more characters from the letter or number categories of Unicode. This definition applies to the string as a whole, regardless of any other characters the string might contain.

I think this would ease some of the confusion around what the "non-empty" definition means.

I know this change would require us to go through already published rules and see which of the definitions it should be using. But I am guessing that such a second look could be beneficial.

Comment (WilcoFiers) [reply _rules/SC4-1-1-unique-id.md]: @jkodu are you going to pick that up? What's the plan for this?

File: pages/glossary/non-empty.md [hunk 2]
```diff
@@ -3,15 +3,4 @@ title: Non-empty text string
...
-- **Lt**: Letter, Titlecase
-- **Lo**: Letter, Other
-- **Lm**: Letter, Modifier
-- **Nd**: Number, Decimal Digit
-
-For more details on unicode categories, check out [www.fileformat.info/info/unicode/category/](http://www.fileformat.info/info/unicode/category/index.htm)
-
-  
\ No newline at end of file
+A string of characters is considered *non-empty* if the string contains characters that are not space characters.
```
Comment (carlosapaduarte) [inline pages/glossary/non-empty.md]: To ensure that space characters are perceived unambiguously perhaps you could use the 'Separator, Space' category of Unicode (and provide the link) similarly to what you do in the textual content definition.

Comment (dd8) [reply pages/glossary/non-empty.md]: The HTML spec allows any character in IDs except ASCII whitespace:
https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute
https://infra.spec.whatwg.org/#ascii-whitespace

So non-ASCII whitespace characters like these are valid in IDs
U+00A0 | NO-BREAK SPACE
U+200B | ZERO WIDTH SPACE

I think the shared definition is problematic because it's used in different contexts where the definition of space is different in each context:
- what is treated as space/empty by the HTML parser
- what is treated as space/empty and displayed as blank or zero width by fonts
- what is treated as space/empty and not voiced by assistive tech

File: pages/glossary/non-empty.md [hunk 3]
```diff
@@ -3,15 +3,4 @@ title: Non-empty text string
 key: non-empty
```
Comment (WilcoFiers) [inline pages/glossary/non-empty.md]: You can't just change this definition. Changing this changes the meaning of a whole bunch of existing rules. If you want to change the meaning of this definition you'll need to create a different definition for the rules that currently use this. Please do that in a separate PR so that we can have a record of it.

File: pages/glossary/non-empty.md (hunk 2 above)
Comment (WilcoFiers) [reply pages/glossary/non-empty.md]: Agreed, "space characters" is insufficient. Please specify which characters are considered space characters.

File: pages/glossary/textual-content.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+title: Textual content
+key: textual-content
+---
+
+A string of characters is considered *textual content* if it contains one or more characters from the letter or number categories of Unicode. This definition applies to the string as a whole, regardless of any other characters the string might contain.
```
Comment (WilcoFiers) [inline pages/glossary/textual-content.md]: Specify the ranges.

**Comment by jkodu** [comment]
There are some definition changes, that anne has picked up https://github.com/auto-wcag/auto-wcag/pull/430/files, hence rolling back definition changes on this PR & keeping a simple change to applicability

**Comment by annethyme** [comment]
@Brynanders , I am assigning this rule to you since you are the owner of this rule inside the WAI-Tools project. It has a deadline of March 31.

File: _rules/SC4-1-1-unique-id.md [hunk 5]
```diff
@@ -20,7 +20,7 @@ authors:
 
 ### Applicability
 
-Any `id` attribute specified on an HTML or SVG element.
+Any `id` attribute which is not an empty string, specified on an HTML or SVG element.
```
Comment (annethyme) [inline _rules/SC4-1-1-unique-id.md]: > which is not an empty string

---> 

> which is not the empty string (""),

Comment (annethyme) [reply _rules/SC4-1-1-unique-id.md]: ... to align perfectly with https://github.com/auto-wcag/auto-wcag/pull/430/files
